sklearn/model_selection/tests/test_search.py

```diff
@@ -139,8 +139,9 @@ def assert_grid_iter_equals_getitem(grid):
       "(key='foo', value=0)")]
 )
 def test_validate_parameter_grid_input(input, error_type, error_message):
-    with pytest.raises(error_type, message=error_message):
+    with pytest.raises(error_type) as raised_error:
         ParameterGrid(input)
+    assert str(raised_error.value) == error_message
```

Comment:
here we want to test the exact message, and the message contains characters that can be parsed in regexp (like `(`),  so I thought this was the way to go

Reply:
Maybe use `match` here?

Reply:
I didn't use it to avoid putting backslashes in front of parenthesis, and to be more restrictive (exact match vs partial match), but I guess indeed it is simpler with match, and match seems more used in the rest of scikit-learn